codec-http2/src/main/java/io/netty/handler/codec/http2/UniformStreamByteDistributor.java

```diff
@@ -179,15 +168,9 @@ void write(int numBytes, Writer writer) throws Http2Exception {
         void addToQueue() {
             if (!enqueued) {
                 enqueued = true;
-                if (streamableBytes == 0) {
-                    // Add empty frames to the empty frame queue.
-                    getOrCreateEmptyFrameQueue().addLast(this);
-                    return;
-                }
-
-                if (!previouslyOnMainQueue) {
-                    // Add to the head the first time it's on the main queue.
-                    previouslyOnMainQueue = true;
+                if (!previouslyEnqueued) {
```

Comment:
Yup, you're right.  Because of headers we always need to add to the end of the queue. Done.